algorithms/integration/kapton_2019_correction.py

```
@@ -0,0 +1,538 @@
+                pt_1=col(edges[ii])
+                pt_2=col(edges[jj])
+                distance=(pt_1-pt_2).length()
+                dlist.append(distance)
+                dlist_idx.append((ii,jj))
+        sorted_idx=flex.sort_permutation(dlist, reverse=True)
+
+        edge_pts=[edges[dlist_idx[sorted_idx[0]][0]], edges[dlist_idx[sorted_idx[1]][0]], edges[dlist_idx[sorted_idx[0]][1]], edges[dlist_idx[sorted_idx[1]][1]]]
+
+        self.detector_edges=edge_pts
+       # Now get the maximum extent of the intersection of the rays with the detector
+        all_ints=[]
+        kapton_path_list = []
+
+        def not_present_in_list(mylist, datapoint, tolerance):
+            """Take a list of tuples and checks if supplied datapoint tuple is within a tolerance limit of each entry in mylist"""
+            if sum([(col(xx)-col(datapoint)).length() <= tolerance for xx in mylist]) ==0:
+                return True
+            else:
+                return False
+
+        for ii,edge_point in enumerate(self.edge_points):
+            s1=edge_point.normalize()
+            kapton_path_mm=self.get_kapton_path_mm(s1)
+            for panel in detector:
+                try:
+                    px_f, px_s=panel.get_pixel_size()
+                    fast_int_px, slow_int_px=panel.get_ray_intersection_px(s1)
+                    x_int, y_int=panel.get_lab_coord(panel.get_ray_intersection(s1))[0:2]
+                    int_point=(x_int, y_int, detz)
+                    # Arbitrary tolerance of couple of pixels otherwise these points were getting clustered together
+                    tolerance=min(panel.get_pixel_size())*2.0
+                    if not_present_in_list(all_ints, int_point, tolerance):
+                        all_ints.append(int_point)
+                    kapton_path_list.append(kapton_path_mm)
+                    #all_ints.append(edge_point)
+                except Exception as e:
+                    if 'DXTBX_ASSERT' not in str(e):
```

This catchall exceptions just swallows the error and print - a) is this really the intended behaviour and b) This should use the (existing) logger instead of printing